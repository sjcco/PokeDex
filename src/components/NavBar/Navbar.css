
.nav {
  display: flex;
  width: 100%;
  background-color: black;
  justify-content: space-between;
}

.nav-item {
  font-size: 1.75rem;
  color: white;
  margin: auto 1rem;
  text-align: center;
  background-color: transparent;
  border: none;
}

.nav-item a {
  color: inherit;
}

.title {
  width: 40%;
  height: 50px;
  margin: 0.3rem 0;
}
