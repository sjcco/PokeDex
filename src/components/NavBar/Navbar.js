import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { nav, navItem, title } from './NavBar.css';
import pokedex from '../../assets/images/title.png';

const NavBar = ({ btn }) => {
  const navBtn = btn => {
    if (btn === 'filter') {
      return (<div className={navItem}><i className="fas fa-filter" /></div>);
    }
    return (
      <div className={navItem}>
        <Link to="/"><i className="fas fa-arrow-left" /></Link>
      </div>
    );
  };

  return (
    <div className={nav}>
      <div className={navItem}><i className="fas fa-bars" /></div>
      <img src={pokedex} alt="..." className={title} />
      {navBtn(btn)}
    </div>
  );
};

NavBar.propTypes = {
  btn: PropTypes.string,
};

NavBar.defaultProps = {
  btn: 'filter',
};

export default NavBar;
