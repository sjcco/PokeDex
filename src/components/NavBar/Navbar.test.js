/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

test('renders content', () => {
  const fnc = jest.fn();
  const component = renderer.create(<Navbar
    btn="filter"
    openFilter={fnc}
  />).toJSON();
  expect(component).toMatchSnapshot();
});
