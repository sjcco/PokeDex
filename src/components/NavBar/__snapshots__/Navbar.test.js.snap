// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content 1`] = `
<div
  className="nav"
>
  <div
    className="navItem"
  >
    <i
      className="fas fa-bars"
    />
  </div>
  <img
    alt="..."
    className="title"
    src="https://i.ibb.co/Th6HjDF/title.png"
  />
  <button
    className="navItem"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-filter"
    />
  </button>
</div>
`;
